src/pages/RFOX/hooks/useGetUnstakingRequestQuery.ts

```diff
@@ -0,0 +1,100 @@
+import { skipToken } from '@tanstack/react-query'
+import { foxStakingV1Abi } from 'contracts/abis/FoxStakingV1'
+import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
+import { useMemo } from 'react'
+import { type Address, getAddress, type MulticallReturnType } from 'viem'
+import { multicall } from 'viem/actions'
+import { arbitrum } from 'viem/chains'
+import type { Config } from 'wagmi'
+import { type ReadContractsQueryKey, useQuery } from 'wagmi/query'
+import { isSome } from 'lib/utils'
+import { viemClientByNetworkId } from 'lib/viem-client'
+
+import { useGetUnstakingRequestCountQuery } from './useGetUnstakingRequestCountQuery'
+
+type AllowFailure = false
+
+const getContracts = (stakingAssetAccountAddress: string | undefined, count: bigint) =>
+  stakingAssetAccountAddress
+    ? Array.from(
+        { length: Number(count) },
+        (_, index) =>
+          ({
+            abi: foxStakingV1Abi,
+            address: RFOX_PROXY_CONTRACT_ADDRESS,
+            functionName: 'getUnstakingRequest',
+            args: [getAddress(stakingAssetAccountAddress), BigInt(index)],
+            chainId: arbitrum.id,
+          }) as const,
+      )
+    : []
+
+type GetContractsReturnType = ReturnType<typeof getContracts>
+
+type GetUnstakingRequestQueryKey = ReadContractsQueryKey<[Address, bigint], AllowFailure, Config>
+
+type UnstakingRequest = MulticallReturnType<GetContractsReturnType, AllowFailure>
+
+type UseGetUnstakingRequestQueryProps<SelectData = UnstakingRequest> = {
+  stakingAssetAccountAddress: string | undefined
+  select?: (unstakingRequest: UnstakingRequest) => SelectData
+}
+
+const client = viemClientByNetworkId[arbitrum.id]
+
+export const useGetUnstakingRequestQuery = <SelectData = UnstakingRequest>({
+  stakingAssetAccountAddress,
+  select,
+}: UseGetUnstakingRequestQueryProps<SelectData>) => {
+  const { data: unstakingRequestCountResponse } = useGetUnstakingRequestCountQuery({
+    stakingAssetAccountAddress,
+  })
+
+  const contracts = useMemo(
+    () =>
+      stakingAssetAccountAddress
+        ? Array.from(
+            { length: Number(unstakingRequestCountResponse) },
+            (_, index) =>
+              ({
+                abi: foxStakingV1Abi,
+                address: RFOX_PROXY_CONTRACT_ADDRESS,
+                functionName: 'getUnstakingRequest',
+                args: [getAddress(stakingAssetAccountAddress), index],
+                chainId: arbitrum.id,
+              }) as const,
+          )
+        : [],
+    [stakingAssetAccountAddress, unstakingRequestCountResponse],
+  )
```

Comment:
Could reuse this guy if we feel like it

```suggestion
  const contracts = useMemo(
    () => getContracts(stakingAssetAccountAddress, unstakingRequestCountResponse ?? 0n),
    [stakingAssetAccountAddress, unstakingRequestCountResponse],
  )
```

Reply:
Oh whoops, I was meant to commit that when I added the fn above - good catch ser!

Reply:
gm ser https://github.com/shapeshift/web/pull/7114/commits/f2671f9c11cff835253cadc76c1ffa61aaba6091